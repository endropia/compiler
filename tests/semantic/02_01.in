var
	i: integer;
	a: array[0..100] of Integer;
	b: record x, y: integer; end;
begin
	for i := 0 to 10 do begin end;
	for i := 3 * 6 + 7 to (3 - 2) - 3 do begin end;
	for a[0] := 0 to 10 do begin end;
	for b.x := 0 to 10 do begin end;

	while (i < 10) do begin
    	i += 1;
    end;

end.