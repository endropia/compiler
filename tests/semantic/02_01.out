program : Unnamed program
   var: 
      i
      type: integer
   var: 
      a
      array
      type: Integer
      range
         0
         100
   var: 
      b
      record
      x
         type: integer
      y
         type: integer
   stmts:
      for
         i
         to
            0
            10
               stmts:
                  empty      for
         i
         to
            +
               *
                  3
                  6
               7
            -
               -
                  3
                  2
               3
               stmts:
                  empty      for
         array
            a
            0
         to
            0
            10
               stmts:
                  empty      for
         b
            x
         to
            0
            10
               stmts:
                  empty      while
         <
            i
            10
            stmts:
               +=
                  i
                  1

scope     name                          class               
------------------------------------------------------------
0         integer                       primitive type      
0         double                        primitive type      
0         boolean                       primitive type      
0         char                          primitive type      
0         string                        primitive type      
1         i                             variable            
1         a                             variable            
1         b                             variable            
